osu.Game/Rulesets/Objects/SliderPath.cs

```diff
@@ -330,6 +330,20 @@ private List<Vector2> calculateSubPath(ReadOnlySpan<Vector2> subControlPoints, P
                     if (subControlPoints.Length != 3)
                         break;
 
+                    CircularArcProperties circularArcProperties = new CircularArcProperties(subControlPoints);
+
+                    //if false, we'll end up breaking anyways when calculating subPath
+                    if (!circularArcProperties.IsValid)
+                        break;
+
+                    //Coppied from PathApproximator.CircularArcToPiecewiseLinear
+                    int subPoints = (2f * circularArcProperties.Radius <= 0.1f) ? 2 : Math.Max(2, (int)Math.Ceiling(circularArcProperties.ThetaRange / (2.0 * Math.Acos(1f - (0.1f / circularArcProperties.Radius)))));
+
+                    //if the amount of subpoints is int.MaxValue, causes an out of memory issue, so we default to bezier
+                    //this only occurs for very large radii, so the result should be essentially a straight line anyways
+                    if (subPoints == int.MaxValue)
+                        break;
```

Comment:
Yeah so this looks flimsy at best to me (but so does the framework implementation of the same thing, for that matter).

- Why is `int.MaxValue` the bound here? You could probably just as easily cause an OOM with half of that number of points. I feel like this bound should be much smarter than that.
- There's an `(int)` cast in that expression there. What guarantees that it doesn't overflow? And yes if it does, then the `Math.Max(2, ...)` t hing probably saves this, but it also makes the huge arc into a segment which also seems not correct.

Reply:
Would agree with it being somewhat flimsy, but just copying framework is the simplest way I could think of to solve this without potential for edge cases. 

From testing, seems to be either a reasonable number or `int.MaxValue`, with nothing really in between. 

I believe this is as for when the radius is greater than ~3 million, arccos(1-0.1/radius) will be less than 0.0003. 
Due to a floating point error, this results in arccos returning `1/double.MaxValue` (presumably, but either way, a value less than `1/int.MaxValue`). Then we divide ThetaRange by this value, which results in a value above `int.MaxValue`. When we cast this double to an integer value, as it's greater than the integer limit, it's instead set to the max value.

Which... does feel very jank, thinking about it. But Framework is going through the same calculation, and we're just making sure that we can't calculate a perfect curve with `int.MaxValue` subpoints regardless regardless.  If it underflows, framework will underflow as well, which also works out fine... ish.

This is also explanation for why it's meaningless to have a lower bound - 12000 sub points is about the maximum you can have without said floating point error.

Reply:
All of the above may be well and true but it's still relying on a bunch of internal implementation details of fp math and arccos and so on. A sensibly defined bound protects against any of that changing around.

Reply:
Ah wait misread your comment initially... Okay, yeah I'd agree. Hmm... did mention 12000 is theoretical max, but probably should be lower still.

I'll go with 1000, we'd need an arc length of greater than ~20 thousand to surpass this, which should be impossible for ranked beat maps anyways.
[desmos link](https://www.desmos.com/calculator/ktmm1iph3f)

Reply:
@smoogipoo can you look at this thing and chip in as to whether you think this sort of check should live in game or maybe it should go live in framework before I start queueing diffcalcs and such to check for regressions?

Reply:
I think this is fine to live in the game